packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue

```diff
@@ -235,22 +256,28 @@ function getModel() {
 }
 
 function saveAsNewModel() {
-	const messageContent = {
-		header: {
-			description: modelName.value,
-			name: modelName.value,
-			_type: 'Header',
-			model_version: 'v1.0',
-			schema: 'modelreps.io/DecaExpr',
-			schema_name: 'DecaExpr'
-		}
-	};
-
-	modelValid.value = false;
+	if (selectedModelFramework.value === ModelFramework.Petrinet) {
+		// save function here for petrinet
+	}
 
-	kernelManager
-		.sendMessage('save_amr_request', messageContent)
-		?.register('save_amr_response', handleSaveAmrResponse);
+	if (selectedModelFramework.value === ModelFramework.Decapodes) {
+		const messageContent = {
+			header: {
+				description: modelName.value,
+				name: modelName.value,
+				_type: 'Header',
+				model_version: 'v1.0',
+				schema: 'modelreps.io/DecaExpr',
+				schema_name: 'DecaExpr'
+			}
+		};
+
+		modelValid.value = false;
+
+		kernelManager
+			.sendMessage('save_amr_request', messageContent)
```

Comment:
Why is the AMR saved via the kernel?

Reply:
saving for petrinet will happen in another PR - This save message is just for the decapodes flow not the petrinet

Reply:
I agree, still why is it happening within the Kernel?

Reply:
I couldnt tell you why its happening that way with the kernel, maybe something for @mwdchang to answer?  Are you saying you would want get the kernel response and then save it with a REST call?

Reply:
in any case I feel that change would be better in a followup PR as well